### PR TITLE
[GCChecker] support c++ code

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

File: Make.inc [hunk 1]
```diff
@@ -507,7 +507,7 @@ $(error Address Sanitizer only supported with clang. Try setting SANITIZE=0)
 endif
 CC  := icc
 CXX := icpc
-JCFLAGS := -std=gnu11 -pipe $(fPIC) -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -fp-model precise -fp-model except -no-ftz
+JCFLAGS := -std=gnu99 -pipe $(fPIC) -fno-strict-aliasing -D_FILE_OFFSET_BITS=64 -fp-model precise -fp-model except -no-ftz
```
Comment (Keno) [inline Make.inc]: Why this change?

Comment (vtjnash) [reply Make.inc]: Slipped in I guess: I had changed them all to gnu11 temporarily, then back to gnu99. Guess I should put this back (https://github.com/JuliaLang/julia/pull/12895).
